### PR TITLE
Add .browserslistrc and update docs around browser support

### LINKED ISSUE #3396: MsalProvider gives Content Security Policy directive: "script-src 'self'" error
### Core Library

@azure/msal-browser

### Core Library Version

2.12.1

### Wrapper Library

@azure/msal-react

### Wrapper Library Version

1.0.0-beta.1

### Description

When I add `<MsalProvider instance={msalInstance}>` I get a "Content Security Policy" error and my app is not loading anymore.
Locally it works fine, but after build and serve it on the Nginx docker container in Azure infrastructure I get this error. If I remove this context initialization the app starts to load again.

### Error Message

runtime.js:747 Uncaught EvalError: Refused to evaluate a string as JavaScript because 'unsafe-eval' is not an allowed source of script in the following Content Security Policy directive: "script-src 'self'".

### Msal Logs

The app is crashing before it is even loaded

### MSAL Configuration

```javascript
{
    auth: {
        clientId: "my-client-id",
        authority: b2cPolicies.authorities.trustFramework.authority,
        knownAuthorities: [b2cPolicies.authorityDomain],
        redirectUri: `${uri}/login`,
    },
    cache: {
        storeAuthStateInCookie: isIE || isEdge || isFirefox
    }
};
```

### Relevant Code Snippets

```javascript
const msalInstance = new PublicClientApplication(msalConfig);
function App() {

  return (
    <MsalProvider instance={msalInstance}>
      <Provider store={store}>
        <ThemeProvider theme={lightTheme}>
          <AppContainer />
        </ThemeProvider>
      </Provider>
    </MsalProvider>
  );
}

ReactDOM.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>,
  document.getElementById('root')
);
```

### Reproduction Steps

1. Add `<MsalProvider instance={msalInstance}>` as shown in examples and the app is not loading

### Expected Behavior

It should not crash an app

### Identity Provider

Azure B2C Custom Policy

### Browsers Affected (Select all that apply)

Chrome, Safari

### Regression

_No response_

### Source

External (Customer)

### PR DESCRIPTION
**Pull request by tnorling**
- Updates docs to explicitly define what browsers we support
- Moves IE 11 caveats/known-issues from wiki to new markdown doc
- Adds `.browserslistrc` defining which browsers we support (this helps prevent certain polyfills for older browsers from being included)
- Adds postinstall step to update the browserslist db with the most up to date browser information (open to feedback around where this should go, it should happen at minimum before releasing)

Closes #3396 by removing polyfill that was causing a conflict with `script-src` CSP.

This was tested and does not break ie11 compatibility as long as the ie11 polyfills are still included in any apps that need to support ie11 (demonstrated in the react-router sample and ie 11 sample).

### CODE REVIEW

**Review by jasonnutter — APPROVED** [review]

File: lib/msal-browser/FAQ.md [hunk 1]
```diff
@@ -59,15 +59,34 @@
...
 
-IE 11, Edge, Chrome, Firefox and Safari
+- Chrome
+- Edge (Chromium)
+- Firefox
+- Safari
+- Opera
 
-Keep these steps in mind when [using MSAL.js with IE](https://github.com/AzureAD/microsoft-authentication-library-for-js/wiki/Using-msal.js-with-Internet-Explorer).
+MSAL.js has also been tested and supports the following browsers with Promise polyfills:
```
Comment (sameerag) [inline lib/msal-browser/FAQ.md:70]: Isn't this contrasting the statement above: "Despite this, IE 11 is currently supported but we do not include Promise polyfills"

Comment (sameerag) [reply lib/msal-browser/FAQ.md:70]: Never mind, is this 1.0?

Comment (tnorling) [reply lib/msal-browser/FAQ.md:70]: This is true for both v1 and v2. We do not include the polyfills, developers need to provide them and this is described in more detail in the IE doc.

Comment (sameerag) [reply lib/msal-browser/FAQ.md:70]: The statement is confusing in that case. Can we always explicitly mention that we do not include polyfills?

Comment (tnorling) [reply lib/msal-browser/FAQ.md]: Updated.

**Review by sameerag — APPROVED** [review]

**Pull request merged**